pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+quotes.
+
+
+### Example of task
+
+       task:
+          jobs:
+            - name: Execute hw.go
+              commands:
+                - echo $SEMAPHORE_PIPELINE_ID
+                - echo $HOME
+                - echo $SEMAPHORE_GIT_DIR
+                - pwd
+                - ls -al
+                - cat hw.go
+                - echo $VAR1
+                - echo $PI
+          prologue:
+              commands:
+                - checkout
+          env_vars:
+               - name: VAR1
+                 value: Environment Variable 1
+               - name: PI
+                 value: "3.14159"
+
+The indentation level of `prologue`, `epilogue`, `env_vars` and `jobs` properties should be the same.
+
+
+## Jobs
+
+The `jobs` items are essential for each Semaphore 2.0 pipeline because they
+allow you to define the actual commands that you want to execute.
+
+### name
+
+The value of the `name` property is a Unicode string that provides a
+description/name for a particular job. It is considered a good practice to give
+descriptive names to the jobs and the blocks items of a Semaphore 2.0 pipeline.
+
+
+### commands
+
+The `commands` property is an array of strings that holds the UNIX commands
+that will be executed for a job.
+
+
+#### The checkout command
+
+The `checkout` command is a special command offered by Semaphore 2.0 and is
+used for fetching the files of the GitHub repository into the virtual machine
+in order to be able to use them.
+
+
+### cmd_file
+
+the value of the `cmd_file` property is a string that points to a plain text
+file that can be found in the GitHub repository of your Semaphore 2.0 project.
+
+Semaphore 2.0 reads the plain text file indicated by `cmd_file` and creates
+the equivalent job using a `commands` block, which is what is finally executed.
+This means that the `cmd_file` property is replaced before the job is started
+and a virtual machine begins its execution.
+
+Notice that you cannot use both `commands` and `cmd_file` when defining a job.
+
+#### Example of a cmd_file block
+
+    jobs:
+      - name: A name for this job
+         cmd_file: GitHub/file/plain_text_file
+
+
+### Example of Jobs
+
+The general structure of a job when using the `commands` property is as follows:
+
+    jobs:
+       - name: A name for this job
+          commands:
+             - Command 1
+             - Command 2
+
+The general structure of a job when using the `cmd_file` property is as follows:
+
+    jobs:
+      - name: A name for this job
+         cmd_file: path/to/file/plain_text_file
+
+
+## Prologue and Epilogue
+
+
+A `.semaphore/semaphore.yml` configuration file can have a single `prologue`
+and a single `epilogue` property per `task` property. Both `prologue` and
+`epilogue` properties are optional.
+
+
+### The Prologue property
+
+A `prologue` block is used when you want to execute some commands prior to
+a job. This is usually the case with initialization commands that install
```

Comment:
Commands specified as `prologue` as prepended to each jobs that is defined in a task.

Reply:
A `prologue` block in a `task` block is used when you want to execute some commands prior to all the `jobs` of that `task`.
Is this any better?